test/ae_mdw_web/controllers/block_controller_test.exs

```diff
@@ -98,6 +98,91 @@ defmodule AeMdwWeb.BlockControllerTest do
     end
   end
 
+  describe "key-block" do
+    test "it gets blocks by kbi", %{conn: conn, store: store} do
+      kbi = 1
+      decoded_hash = TS.key_block_hash(1)
+
+      store =
+        store
+        |> Store.put(Model.Block, Model.block(index: {kbi - 1, -1}, hash: TS.key_block_hash(0)))
+        |> Store.put(Model.Block, Model.block(index: {kbi, -1}, hash: decoded_hash, tx_index: 4))
+
+      with_mocks [
+        {:aec_db, [], [get_header: fn ^decoded_hash -> :header end]},
+        {:aec_headers, [], [serialize_for_client: fn :header, :micro -> %{height: kbi} end]},
+        {AeMdw.Node.Db, [], [prev_block_type: fn :header -> :micro end]}
+      ] do
+        assert %{
+                 "height" => ^kbi,
+                 "micro_blocks_count" => 0,
+                 "transactions_count" => 0
+               } =
+                 conn
+                 |> with_store(store)
+                 |> get("/v2/key-blocks/#{kbi}")
+                 |> json_response(200)
+      end
+    end
+
+    test "it returns a block with txs counts", %{conn: conn, store: store} do
+      kbi = 1
+      decoded_hash = TS.key_block_hash(1)
+      encoded_hash = :aeser_api_encoder.encode(:key_block_hash, decoded_hash)
+
+      store =
+        store
+        |> Store.put(Model.Block, Model.block(index: {kbi - 1, -1}, hash: TS.key_block_hash(0)))
+        |> Store.put(Model.Block, Model.block(index: {kbi, -1}, hash: decoded_hash, tx_index: 4))
+        |> Store.put(
+          Model.Block,
+          Model.block(index: {kbi, 0}, hash: TS.micro_block_hash(0), tx_index: 0)
+        )
+        |> Store.put(
+          Model.Block,
+          Model.block(index: {kbi + 1, -1}, hash: TS.key_block_hash(2), tx_index: 10)
+        )
+
+      with_mocks [
+        {:aec_chain, [], [get_block: fn ^decoded_hash -> {:ok, :block} end]},
+        {:aec_db, [], [get_header: fn ^decoded_hash -> :header end]},
+        {:aec_blocks, [], [to_header: fn :block -> :header end]},
+        {:aec_headers, [],
+         [
+           height: fn :header -> kbi end,
+           serialize_for_client: fn :header, :micro -> %{height: kbi} end,
+           type: fn :header -> :key end
+         ]},
+        {AeMdw.Node.Db, [], [prev_block_type: fn :header -> :micro end]}
+      ] do
+        assert %{
+                 "height" => ^kbi,
+                 "transactions_count" => 6
+               } =
+                 conn
+                 |> with_store(store)
+                 |> get("/v2/key-blocks/#{encoded_hash}")
+                 |> json_response(200)
+      end
+    end
+
+    test "it returns 404 when not found", %{conn: conn, store: store} do
+      decoded_hash = TS.key_block_hash(0)
+      encoded_hash = :aeser_api_encoder.encode(:key_block_hash, decoded_hash)
+      error_msg = "not found: #{encoded_hash}"
+
+      with_mocks [
+        {:aec_chain, [], [get_block: fn ^decoded_hash -> :error end]}
+      ] do
```

Comment:
It is not needed (the ci database won't have it)

Reply:
Done